test/e2e/containerruntimeconfig_test.go

```diff
@@ -0,0 +1,87 @@
+package e2e_test
+
+import (
+	"fmt"
+	"testing"
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/openshift/machine-config-operator/test/e2e/framework"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+func TestContainerRuntimeConfigLogSizeMax(t *testing.T) {
+	runTestWithCtrcfg(t, "log-size-max", &mcfgv1.ContainerRuntimeConfiguration{
+		LogSizeMax: resource.MustParse("-1"),
+	})
+}
+
+func TestContainerRuntimeConfigPidsLimit(t *testing.T) {
+	runTestWithCtrcfg(t, "pids-limit", &mcfgv1.ContainerRuntimeConfiguration{
+		PidsLimit: 2048,
+	})
+}
+
+func runTestWithCtrcfg(t *testing.T, testName string, cfg *mcfgv1.ContainerRuntimeConfiguration) {
+	cs := framework.NewClientSet("")
+	testUUID := uuid.NewUUID()
+	matchValue := fmt.Sprintf("%s-%s", testName, testUUID)
+	ctrcfgName := fmt.Sprintf("ctrcfg-%s", testUUID)
+
+	// cache the old machine config value to compare against
+	oldMC := getMcName(t, cs, "worker")
+
+	// label one node from the pool to specify which worker to update
+	unlabelFunc := labelRandomNodeFromPool(t, cs, "worker", mcpNameToRole(matchValue))
+	defer unlabelFunc()
+
+	// create an MCP to match the node we tagged
+	mcpCleanupFunc := createMCP(t, cs, matchValue)
+
+	// create our ctrcfg and attach it to our created node pool
+	cleanupCtrcfg := createCtrcfgWithConfig(t, cs, ctrcfgName, matchValue, cfg)
+
+	// wait for the ctrcfg to show up
+	mcName, err := waitForRenderedConfigFromCtrcfg(t, cs, matchValue, ctrcfgName)
+	require.Nil(t, err, "failed to render machine config from container runtime config")
+
+	// wait for the ctrcfg to complete its update
+	err = waitForPoolComplete(t, cs, matchValue, mcName)
+	require.Nil(t, err)
+
+	// cleanup ctrcfg and make sure it doesn't error
+	err = cleanupCtrcfg()
+	require.Nil(t, err)
+
+	t.Logf("Deleted ContainerRuntimeConfig %s", ctrcfgName)
```

Comment:
A proper rollback test would also be good here.

Reply:
proper rollback test added!